pp.c

```diff
@@ -7939,10 +7934,158 @@ PP(pp_argcheck)
         /* diag_listed_as: Odd name/value argument for subroutine '%s' */
         Perl_croak_caller("Odd name/value argument for subroutine '%" SVf "'",
                           S_find_runcv_name());
+}
+
+PP(pp_argcheck)
+{
+    OP * const o       = PL_op;
+    struct op_argcheck_aux *aux = (struct op_argcheck_aux *)cUNOP_AUXo->op_aux;
+    AV  *defav         = GvAV(PL_defgv); /* @_ */
+
+    assert(!SvMAGICAL(defav));
+    UV argc = (UV)(AvFILLp(defav) + 1);
+
+    S_check_argc(aTHX_ argc, aux->params, aux->opt_params, aux->slurpy);
 
     return NORMAL;
 }
 
+PP(pp_multiparam)
+{
+    struct op_multiparam_aux *aux = (struct op_multiparam_aux *)cUNOP_AUX->op_aux;
+    UV nparams = aux->n_positional;
+    char slurpy = aux->slurpy;
+    AV  *defav = GvAV(PL_defgv); /* @_ */
+
+    assert(!SvMAGICAL(defav));
+    UV argc = (UV)(AvFILLp(defav) + 1);
+
+    S_check_argc(aTHX_ argc, nparams, nparams - aux->min_args, slurpy);
+
+    UV parami;
+    for(parami = 0; parami < nparams; parami++) {
+        PADOFFSET padix = aux->param_padix[parami];
```

Comment:
factor out of the loop reading's struct aux's param_padix field over and over , nothing will be reallocing it afaik